torchinductor/codecache.py

```diff
@@ -161,8 +165,9 @@ def load(cls, source_code):
                 mod = types.ModuleType(f"{__name__}.{key}")
                 mod.__file__ = path
                 exec(code, mod.__dict__, mod.__dict__)
-                cls.cache[key] = mod
-                cls.cache[key].key = key
+                if key not in cls.cache:
```

Comment:
Is this not already always the case because of `if key not in cls.cache:` not on L162?

Reply:
This PR adds multithreading.  I've been experimenting with loading these in parallel in a racy way.  If another thread won the race, we want to reuse its result to have fewer compiles.

Reply:
Oh nice, that makes sense. Should we add a comment to that effect?